app/src/main/java/org/schabi/newpipe/util/SecondaryStreamHelper.java

```diff
@@ -41,33 +46,36 @@ public static AudioStream getAudioStreamFor(@NonNull final List<AudioStream> aud
 
         switch (mediaFormat) {
             case WEBM:
-            case MPEG_4:// ¿is mpeg-4 DASH?
+            case MPEG_4: // Is MPEG-4 DASH?
                 break;
             default:
                 return null;
         }
 
-        final boolean m4v = (mediaFormat == MediaFormat.MPEG_4);
+        final boolean m4v = mediaFormat == MediaFormat.MPEG_4;
+        final boolean isLimitingDataUsage = ListHelper.isLimitingDataUsage(context);
+
+        Comparator<AudioStream> comparator = ListHelper.getAudioFormatComparator(
+                m4v ? MediaFormat.M4A : MediaFormat.WEBMA, isLimitingDataUsage);
+        int preferredAudioStreamIndex = ListHelper.getAudioIndexByHighestRank(
+                audioStreams, comparator);
 
-        for (final AudioStream audio : audioStreams) {
-            if (audio.getFormat() == (m4v ? MediaFormat.M4A : MediaFormat.WEBMA)) {
-                return audio;
+        if (preferredAudioStreamIndex == -1) {
+            if (m4v) {
+                return null;
             }
-        }
 
-        if (m4v) {
-            return null;
-        }
+            comparator = ListHelper.getAudioFormatComparator(
+                    MediaFormat.WEBMA_OPUS, isLimitingDataUsage);
+            preferredAudioStreamIndex = ListHelper.getAudioIndexByHighestRank(
+                    audioStreams, comparator);
 
-        // retry, but this time in reverse order
```

Comment:
Why on earth were they searched in reverse order the first time?!?
<sub>(this is not a review comment)</sub>